### PR TITLE
[feat]: Add schema for all supported video platforms

### PR DESCRIPTION
**Pull request by shraddha-kesari**
**Description:** 
Currently, we only support only youtube by passing `embed-url` key to the video schema.

We now need to add support for all embed players supported by Bold. Below is the list of supported embed players with their respective keys in the API:

- [ ]  YouTube: url 
- [ ]  Dailymotion: dailymotion-url
- [ ]  Instagram: instagram-url
- [ ]  Facebook: facebook-url
- [ ]  Twitter: tweet-url
- [ ]  Vimeo: vimeo-url
- [ ]  Brightcove: player-url (Platform needs to fix this URL)

**Ref ticket:** https://github.com/quintype/page-builder/issues/3122

### CODE REVIEW

File: src/structured-data/structured-data-tags.js [hunk 1]
```diff
@@ -263,23 +263,32 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
...
-  // find is used for early exit
-  cards.find((card) => {
+  for (const card of cards) {
     const storyElements = get(card, ["story-elements"], []);
-    return storyElements.find((elem) => {
-      if (elem["embed-url"]) {
-        embedUrl = elem["embed-url"];
-        return true;
+    for (const elem of storyElements) {
+      if (elem.subtype && matchSubtypeWithUrl[elem.subtype]) {
```
Comment (ags1773) [inline src/structured-data/structured-data-tags.js:278]:
```suggestion
      if (elem.subtype && elem.subtype in matchSubtypeWithUrl) {
```

File: src/structured-data/structured-data-tags.js [hunk 2]
```diff
@@ -263,23 +263,32 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
 }
 
 function getEmbedUrl(cards) {
-  let embedUrl = "";
+  const matchSubtypeWithUrl = {
```
Comment (ags1773) [inline src/structured-data/structured-data-tags.js:266]: rename to `playerUrlMapping`

File: src/structured-data/structured-data-tags.js (hunk 1 above)
Comment (ags1773) [inline src/structured-data/structured-data-tags.js:278]:
```suggestion
      if (elem.subtype && elem.subtype in matchSubtypeWithUrl) {
```

File: src/structured-data/structured-data-tags.js [hunk 3]
```diff
@@ -263,23 +263,32 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
...
-  cards.find((card) => {
+  for (const card of cards) {
     const storyElements = get(card, ["story-elements"], []);
-    return storyElements.find((elem) => {
-      if (elem["embed-url"]) {
-        embedUrl = elem["embed-url"];
-        return true;
+    for (const elem of storyElements) {
+      if (elem.subtype && matchSubtypeWithUrl[elem.subtype]) {
+        const key = matchSubtypeWithUrl[elem.subtype];
```
Comment (ags1773) [inline src/structured-data/structured-data-tags.js:279]: call this variable `playerUrlField`

File: src/structured-data/structured-data-tags.js [hunk 4]
```diff
@@ -263,23 +263,32 @@ function generateLiveBlogPostingData(structuredData = {}, story = {}, publisherC
...
+        if (elem.metadata && elem.metadata[key]) {
+          return elem.metadata[key];
+        }
+      } else if (elem.type === "youtube-video" && elem.subtype === null) {
```
Comment (ags1773) [inline src/structured-data/structured-data-tags.js:283]: does this need to be an `else if`?

Comment (shraddha-kesari) [reply src/structured-data/structured-data-tags.js:283]: Need not. Can be an if condition itself. Will change it.

**Review by ags1773 — APPROVED** [review]

**Review by ReenaSingh07 — APPROVED** [review]

**Pull request merged**